editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/flag/FlagListFragment.java

```diff
@@ -54,53 +49,67 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
-    private List<Flag> flagList;
 
-    public class mViewHolder extends RecyclerView.ViewHolder {
-        public TextView name;
-        public TextView description;
-        public ImageView image;
-        public Button editButton;
+class FlagViewHolder extends RecyclerView.ViewHolder {
+    public TextView name;
+    public TextView description;
+    public ImageView image;
+    public Button editButton;
 
-        public mViewHolder(View view) {
-            super(view);
-            name = view.findViewById(R.id.nameTextView);
-            description = view.findViewById(R.id.descriptionTextView);
-            image = view.findViewById(R.id.mImageView);
-            editButton = view.findViewById(R.id.editButton);
+    public FlagViewHolder(View view) {
+        super(view);
+        name = view.findViewById(R.id.nameTextView);
+        description = view.findViewById(R.id.descriptionTextView);
+        image = view.findViewById(R.id.mImageView);
+        editButton = view.findViewById(R.id.editButton);
 
-            editButton.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(View v) {
+        editButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
 
-                }
-            });
+            }
+        });
+    }
+}
+
+
+class FlagViewModel extends ViewModel {
+    private List<Flag> flagList;
+
+    public List<Flag> getFlagList(){
+        if(flagList == null) {
+            //TODO: database query
         }
+        return flagList;
     }
+}
+
 
-    public FlagAdapter (List<Flag> flagList) {
-        this.flagList = flagList;
+class FlagAdapter extends ListAdapter<Flag, FlagViewHolder> {
+
+    public FlagAdapter() {
+        super(FlagAdapter.DIFF_CALLBACK);
     }
 
-    // Tells the recyclerview to use the flag_list_fragment_view layout as the layout for each row in the recyclerview
     @Override
-    public mViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+    public FlagViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
         View itemView = LayoutInflater.from(parent.getContext())
                 .inflate(R.layout.flag_list_fragment_row, parent, false);
 
-        return new mViewHolder(itemView);
+        return new FlagViewHolder(itemView);
     }
 
-    // "Binds" each datapoint from the database to a viewholder in the recyclerview
     @Override
-    public void onBindViewHolder(mViewHolder holder, int position) {
+    public void onBindViewHolder(FlagViewHolder holder, int position) {
+
+        // Gets the list of flags that will be displayed in the list view from the view model
+        FlagViewModel model = ViewModelProviders.of(FlagListFragment.newInstance()).get(FlagViewModel.class);
+        List<Flag> flagList = model.getFlagList();
+
         Flag flag = flagList.get(position);
 
         holder.name.setText(flag.getPanel().getName());
```

Comment:
Move code from 115-126 as a separate function called "bindTo" in FlagViewHolder after the constructor. We're moving it because each class should only do one thing. ViewHolder manages content of a single row and onBindViewHolder binds ViewHolder to RecyclerView. Right now, whenever there's a row content change, both ViewHolder and ListAdapter would need to be aware. We avoid that by moving this code! Then, onBindViewHolder just needs to call bindTo.